pycalphad/property_framework/units.py

```diff
@@ -0,0 +1,79 @@
+import pint
+import numpy as np
+import numpy.typing as npt
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from pycalphad.property_framework import ComputableProperty
+
+ureg = pint.UnitRegistry(preprocessors=[lambda s: s.replace('%', ' percent ')])
+ureg.define('atom = 1/avogadro_number * mol')
+ureg.define('fraction = []')
+ureg.define('percent = 1e-2 fraction = %')
+ureg.define('ppm = 1e-6 fraction')
+pint.set_application_registry(ureg)
+Q_ = ureg.Quantity
+DimensionalityError = pint.DimensionalityError
+
+def as_quantity(prop: "ComputableProperty", qt: npt.ArrayLike):
+    if not isinstance(qt, Q_):
+        return Q_(qt, prop.display_units)
+    else:
+        return qt
+
+energy_implementation_units = GM_implementation_units = 'J / mol'
+energy_display_units = GM_display_units = 'J / mol'
+energy_display_name = GM_display_name = 'Gibbs Energy'
+G_implementation_units = 'J'
+G_display_units = 'J'
+G_display_name = 'Gibbs Energy'
```

Comment:
These are duplicated in `Model`. Also repeating the nit that the `energy_implementation_units` should be `G` not `GM`

Reply:
Addressed in the [other comment](https://github.com/pycalphad/pycalphad/pull/432#discussion_r1677169198)